plugins/org.eclipse.n4js/src/org/eclipse/n4js/naming/ModuleNameComputer.xtend

```diff
@@ -66,7 +66,11 @@ class ModuleNameComputer {
 				} else {
 					relativeURI = URIUtils.trimFileExtension(relativeURI);
 				}
-				return QualifiedName.create(relativeURI.segments)
+				val segments = relativeURI.segments;
+				for (var i = 0; i < segments.length; i++) {
+					segments.set(i, URI.decode(segments.get(i)));
+				}
+				return QualifiedName.create(segments);
```

Comment:
Looks like a candidate for `modifySegments()`, right?

Reply:
At first, I had added this decoding somewhere else, where `modifySegments()` was a better fit. Then I moved the decoding to this place where some unnecessary array copying can be avoided by doing it with a loop. Still, I did not want to delete `modifySegments()` even though it is dead code now, because it is just a very small utility.